src/genius/services/http-provider/http-data-provider.service.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  public addNewPerson(person: Person): Observable<Object> {
+    const httpOptions = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      })
+    };
+    let newPerson = new PersonDTO();
+    newPerson.name = {};
+    if (person.firstName) {
+      newPerson.name.first = person.firstName;
+    }
+    if (person.lastName) {
+      newPerson.name.last = person.lastName;
+    }
+    if (person.middleName) {
+      newPerson.name.middle = person.middleName;
+    }
+    if (person.sex) {
+      newPerson.gender = person.sex.toUpperCase();
+    }
+
+    return this.http.post(`${environment.url}/persons`, newPerson, httpOptions)
+  }
+
+  public changeFamily(family: Family): Observable<Object> {
+    let changeFamily = new FamilyDTO()
+    if (family.father) {
+      changeFamily.husband = family.father.id;
+    }
+    if (family.mother) {
+      changeFamily.wife = family.mother.id;
+    }
+    if (family.children && family.children.length > 0) {
+      changeFamily.children = family.children.map(child => child.id)
+    } else {
+      changeFamily.children = [];
+    }
+
+     return this.http.put(`${environment.url}/families/${family.id}`, changeFamily)
+  }
+
+  public changePerson(person: Person): Observable<Object>{
+    let changePerson = new PersonDTO();
+    changePerson.name = {first: null,}
+    if (person.firstName) {
+      changePerson.name.first = person.firstName;
+    }
+    if (person.lastName) {
+      changePerson.name.last = person.lastName;
+    }
+    if (person.middleName) {
+      changePerson.name.middle = person.middleName;
+    }
+    if (person.sex) {
+      changePerson.gender = person.sex.toUpperCase();
+    }
+
+     return this.http.put(`${environment.url}/persons/${person.id}`, changePerson)
+  }
+
+  public deleteFamily(familyId: number): Observable<Object> {
+    return this.http.delete(`${environment.url}/families/${familyId}`)
+  }
+
+  public deletePerson(personId: number): Observable<Object> {
+    return this.http.delete(`${environment.url}/persons/${personId}`)
+  }
+
+  public findFamily(familyId: number): Observable<Family> {
+    return this.http.get<Object>(`${environment.url}/persons/${familyId}`)
+      .pipe(
+        map(httpResponse => this.mapFamily(httpResponse))
+      )
+  }
+
+  public findPerson(personId: number): Observable<Person> {
+    return this.http.get<Object>(`${environment.url}/persons/${personId}`)
+      .pipe(
+        map(httpResponse => this.mapPerson(httpResponse))
+      )
+  }
+
+  public getFamilies(): Observable<Array<Family>> {
+    return this.http.get<Array<Object>>(`${environment.url}/families`)
+      .pipe(
+        map(httpResponse => httpResponse.map(obj => this.mapFamily(obj)))
+      );
+  }
+
+  public getPersons(): Observable<Array<Person>> {
+    return this.http.get<Array<Object>>(`${environment.url}/persons`)
```

Comment:
Client must specify what data type it expects. All get http requests must specify data type server must return. 
'Accept': 'application/json' is mandatory for get requests